packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -224,11 +286,19 @@ export async function sideloadAddIn(manifestPath: string, app?: OfficeApp, canPr
     }
   }
 
-  await registerAddIn(manifestPath);
-
-  if (app) {
-    const sideloadFile = await generateSideloadFile(app, manifest);
-
-    await open(sideloadFile, { wait: false });
+  if (isSideloadingSupportedForHost(app, isDesktop)) {
+    if (isDesktop) {
+      await registerAddIn(manifestPath);
+      sideloadFile = await generateSideloadFile(app, manifest, document);
+      await open(sideloadFile, { wait: false });
+    } else {
+      if (document === undefined) {
```

Comment:
Rather than check for document being defined here, you can pass it to generateSideloadUrl() and it can do the validation.

A better error message would be "The document url was not provided."  

Remember that you can get errors back when importing the function rather than using the command line so it is better to not reference the command line switch.